make/README.md

```diff
@@ -1,94 +1,114 @@
+## kind-version prints the version of kind.
+##
+## Bet you didn't expect that, huh?
+##
+## @category Development
+kind-version: | $(BINDIR)/tools/kind
+	@$(BINDIR)/tools/kind --version
 ```
 
-> **Tip:** this change of PATH won't persist between shell sessions. To get
-> this command executed automatically when you enter the cert-manager
-> folder, put this command in an `.envrc` file in the cert-manager folder
-> and install [`direnv`](https://direnv.net/docs/installation.html).
+Categories are loosely defined; check the output of `make help` for examples of the kinds of categories we already have.
 
-## Common usages
+Regular comments above a target should start with a single `#`.
 
-This section describes common usage patterns for development and testing.
+### Should it be `.PHONY`?
 
-### Installing a development build of cert-manager
+> A phony target is one that is not really the name of a file; rather it is just a name for a recipe to be executed when you make an explicit request.
 
-Once you have a kind cluster running, you can install a development version of
-cert-manager by running:
+The [GNU Make documentation](https://www.gnu.org/software/make/manual/html_node/Phony-Targets.html) gives the above definition for `.PHONY` targets, and is
+worth reading if you're adding a new target since getting this wrong can lead to either spurious rebuilds or unexpected failures to execute your target.
 
-```sh
-make -j8 e2e-setup-certmanager
-```
+Put short: If a target doesn't create a file with the same name as the target, it should be `.PHONY`.
 
-This will create a kind cluster, build, load and install cert-manager from
-source into your kind development cluster.
+Mark the target as `.PHONY` by adding the declaration directly above the target, with any documentation comments in between. For example:
 
-Further invocations of this command will rebuild and upgrade the installed
-version of cert-manager, making it possible to iteratively work on the
-codebase and test changes.
+```make
+.PHONY: my-target
+## Does something awesome!
+##
+## @category Awesomeness
+my-target:
+	@echo Something awesome!
+```
 
-### Running end-to-end tests
+### Target Dependencies / Prerequisites
 
-Before running the end-to-end tests, you must install some additional
-components used during the tests into your kind cluster.
+Make has [two types of dependency](https://www.gnu.org/software/make/manual/html_node/Prerequisite-Types.html): "normal" and "order-only".
 
-Run the following to setup cert-manager, Pebble, ingress-nginx, the sample
-DNS01 webhook and all the other components required for the end-to-end
-tests:
+When creating or changing a target, you should choose the type based on what your dependency is.
 
-```sh
-make -j8 e2e-setup
-```
+⚠️ If your dependency is a `.PHONY` target, you should think very hard about whether to include it. A `.PHONY` dependency will force your target to be rebuilt every single time. That's rarely what you want.
```

Comment:
Aha, that's a good gotcha to document.. so if you make a PHONY dependencies a "normal" pre-requisite (like most people probably do), your target will always be execute unconditionally 👀